### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by eMrazSVK**
New features requests:

- Cherry pick from various branches into devel and the PR into main
- Finish oem branch PR and cherry pick what we want from it to devel
- Everything  done is merged into jp512_20_configs
---

Feature list:

- Orin NX, Orin NX 8GB, Orin Nano support → requires code adjustments too, this is in jp512_20_configs - there are code changes, but configs are not present yet.
- Set boot partition (drive) to use UUID instead of partition name. →  Add readme about ab_partitioning → experimental feature …
- Consider adding local overlays (local_overlays branch)
- Add mavlink-router installation (using binaries) and copy default config for it and also create systemd service.
- Service cleanup (usb3, usb-hub, eth switch).
- Check if default nvpmodel can be set in device tree
- Fan control - in custom_rootfs_and_batch branch.
- Anything above mavlink-router will be included in sample rootfs provided by us.
- Add wifi generator script to resources. (inside provided rootfs)
- Add FMU script (switching between CUBE and onboard FMU). (inside provided rootfs).
- Document each feature.
- Correct permissions and UDEV rules → inside custom_rootfs_and_batch, dcs_first_boot.sh
- Purge SSH keys → inside custom_rootfs_and_batch.

### CODE REVIEW

File: dcs_deploy.py [hunk 1]
```diff
@@ -7,17 +7,53 @@
 import wget
 from threading import Thread, Event
 import time
+from urllib.parse import urlparse
+import sys as _sys
+
+dcs_deploy_version = "0.1.0"
```
Comment (eMrazSVK) [inline dcs_deploy.py:13]: Maybe we should use main version `2` if we call it release 2?
```suggestion
dcs_deploy_version = "2.1.0"
```

File: dcs_deploy.py [hunk 2]
```diff
@@ -446,31 +703,131 @@ def load_selected_config(self):
...
+                # setup multiple app partitions
+                self.ext_partition_layout = os.path.relpath('tools/kernel_flash/flash_l4t_nvme_rootfs_ab.xml')
+            else:
+                # setup no multiple app partitions
+                self.ext_partition_layout = os.path.relpath('tools/kernel_flash/flash_l4t_external_custom.xml')
+        else:
+            print("Unknown storage [%s]! exitting" % self.config['storage'])
+            exit(9)
+        # fix default rootdev to external  (or internal) for orin. There is NFS used to flash
+        if self.config['device'] == 'orin_nx' or self.config['device'] == 'orin_nx_8gb' or self.config['device'] == 'orin_nano_8gb' or self.config['device'] == 'orin_nano_4gb':
```
Comment (eMrazSVK) [inline dcs_deploy.py:757]: More compact and more readable version
```suggestion
        if self.config['device'] in ['orin_nx', 'orin_nx_8gb', 'orin_nano_8gb', 'orin_nano_4gb']:
```

File: dcs_deploy.py [hunk 3]
```diff
@@ -446,31 +703,131 @@ def load_selected_config(self):
...
+                else:
+                    opt_app_size = 8
+                opt_app_size_arg = f"-S {opt_app_size}GiB"
+                external_only = "" # flash internal and external device
+                #self.rootdev = "external" # set UUID device in kernel commandline: rootfs=PARTUUID=<external-uuid>
+
+            if self.args.app_size is not None:
+                opt_app_size_arg = f"-S {self.args.app_size}GiB"
+
+            if self.config['device'] == 'orin_nx' or self.config['device'] == 'orin_nx_8gb' or self.config['device'] == 'orin_nano_8gb' or self.config['device'] == 'orin_nano_4gb':
```
Comment (eMrazSVK) [inline dcs_deploy.py:796]: More compact and more readable version nr. 2
```suggestion
            if self.config['device'] in ['orin_nx', 'orin_nx_8gb', 'orin_nano_8gb', 'orin_nano_4gb']:
```

File: readme.md [hunk 4]
```diff
@@ -79,15 +136,40 @@ As a root of this filesystem, `.dcs_deploy` folder is created inside **host pc H
...
+TcpServerPort = 0
+# Leave to TCP if using SITL
+# Main connection to AutoPilot
+[UartEndpoint cube]
+Device = /dev/ttyTHS0
+Baud = 921600
+
+[UdpEndpoint GCS]
+Mode = Normal
+Address = 10.223.255.255
```
Comment (eMrazSVK) [inline readme.md:163]:
```suggestion
Address = 192.168.55.100
```

File: readme.md [hunk 5]
```diff
@@ -79,15 +136,40 @@ As a root of this filesystem, `.dcs_deploy` folder is created inside **host pc H
...
+[General]
+DebugLogLevel = info
+TcpServerPort = 0
+# Leave to TCP if using SITL
+# Main connection to AutoPilot
+[UartEndpoint cube]
+Device = /dev/ttyTHS0
+Baud = 921600
+
+[UdpEndpoint GCS]
```
Comment (eMrazSVK) [inline readme.md:161]:
```suggestion
# Note that this is for communication through dev micro USB. 
# if you want to use any other interface, change address accordingly. 
[UdpEndpoint GCS]
```

**Review by eMrazSVK — COMMENTED** [review]
I've added some suggestions for @MartinSedlacek work in this PR. Now I'm waiting for review on my changes.

File: readme.md [hunk 6]
```diff
@@ -79,15 +136,42 @@ As a root of this filesystem, `.dcs_deploy` folder is created inside **host pc H
...
+Address = 192.168.55.100
+Port = 14550
+``` 
+
 ### Known limitations
 - When the script is re-ran, flash config folder is deleted and the files are extracted again.
-  - If user customize files in flash config folder, this changes are not saved for next flashing using `dcs-deploy` script.
-  - This adds ~3 min time to the whole process.
 - The database of configs is held inside this repository, which is not ideal.
 - Limited config suppport (only Xavier NX + JP51, emmc or nvme flashing at the moment)
```
Comment (MartinSedlacek) [inline readme.md:172]: This can be deleted.
```suggestion
```

**Pull request merged**